File: test-suite/src/test/java/io/micronaut/docs/http/bind/binders/ShoppingCart.java
package io.micronaut.docs.http.bind.binders;
// tag::class[]
import io.micronaut.core.annotation.Introspected;

@Introspected
public class ShoppingCart {
    private String sessionId;
    private Integer total;

    public String getSessionId() {
        return sessionId;
    }

    public void setSessionId(String sessionId) {
        this.sessionId = sessionId;
    }

    public Integer getTotal() {
        return total;
    }

    public void setTotal(Integer total) {
        this.total = total;
    }
}
// end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/http/server/bind/annotation/ShoppingCart.kt
package io.micronaut.docs.http.server.bind.annotation

// tag::class[]
import io.micronaut.core.bind.annotation.Bindable

@Target(AnnotationTarget.FIELD, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.ANNOTATION_CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Bindable //<1>
annotation class ShoppingCart(val value: String = "")
// end::class[]
